sound/soc/sof/intel/hda-loader.c

```diff
@@ -306,49 +321,38 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 	if (i == HDA_FW_BOOT_ATTEMPTS) {
 		dev_err(sdev->dev, "error: dsp init failed after %d attempts with err: %d\n",
 			i, ret);
-
-		hda_dsp_dump(sdev, SOF_DBG_REGS | SOF_DBG_PCI | SOF_DBG_MBOX);
-
-		/* disable DSP */
-		snd_sof_dsp_update_bits(sdev, HDA_DSP_PP_BAR,
-					SOF_HDA_REG_PP_PPCTL,
-					SOF_HDA_PPCTL_GPROCEN, 0);
-		goto done;
+		goto cleanup;
```

Comment:
you could also do the loop to find a stream before the error handling and jump to cleanup, that way you do it only once?

Reply:
@plbossart good idea. fixed now.